roaring/tests/iter_advance_to.rs

```diff
@@ -0,0 +1,79 @@
+use roaring::RoaringBitmap;
+
+#[test]
+fn iter_basic() {
+    let bm = RoaringBitmap::from([1, 2, 3, 4, 11, 12, 13, 14]);
+    let mut i = bm.iter().advance_to(10);
+    for n in 11..=14 {
+        assert_eq!(i.next(), Some(n))
+    }
+    assert_eq!(i.next(), None);
+}
+
+#[test]
+fn iter_advance_past_end() {
+    let bm = RoaringBitmap::from([1, 2, 3, 4, 11, 12, 13, 14]);
+    let mut i = bm.iter().advance_to(15);
+    assert_eq!(i.next(), None);
+}
+
+#[test]
+fn iter_multi_container() {
+    let bm = RoaringBitmap::from([1, 2, 3, 100000, 100001]);
+    let mut i = bm.iter().advance_to(3);
+    assert_eq!(i.next(), Some(3));
+    assert_eq!(i.next(), Some(100000));
+    assert_eq!(i.next(), Some(100001));
+    assert_eq!(i.next(), None);
+}
+
+#[test]
+fn iter_empty() {
+    let bm = RoaringBitmap::new();
+    assert_eq!(bm.iter().advance_to(31337).next(), None)
+}
+
+#[test]
+fn into_iter_basic() {
+    let bm = RoaringBitmap::from([1, 2, 3, 4, 11, 12, 13, 14]);
+    let mut i = bm.into_iter().advance_to(10);
+    for n in 11..=14 {
+        assert_eq!(i.next(), Some(n))
+    }
+    assert_eq!(i.next(), None);
+}
+
+#[test]
+fn into_iter_multi_container() {
+    let bm = RoaringBitmap::from([1, 2, 3, 100000, 100001]);
+    let mut i = bm.into_iter().advance_to(3);
+    assert_eq!(i.next(), Some(3));
+    assert_eq!(i.next(), Some(100000));
+    assert_eq!(i.next(), Some(100001));
+    assert_eq!(i.next(), None);
+}
+
+#[test]
+fn into_iter_empty() {
+    let bm = RoaringBitmap::new();
+    assert_eq!(bm.into_iter().advance_to(31337).next(), None)
+}
+
+#[test]
+fn iter_from() {
+    let bm = RoaringBitmap::from([1, 2, 3, 100000, 100001]);
+    let mut i = bm.iter_from(99999);
```

Comment:
What is the point of having an `iter_from` method when we can just use the `.advance_to` one on an iterator?

Reply:
@Kerollmops thanks for the review, I'll take a closer look at it tomorrow. As for this particular thing, my use case has very large bitmaps and extreme throughput requirements so I added this function because of that. It uses a binary search instead of the looping done in the iterators, which gains some speed. I will point out the difference between iter_from and iter().advance_to in the documentation

Reply:
Nice and wouldn't it be possible to construct a normal iterator i.e., `Iter` or `IntoIter`, with this binary search technique instead of having a special one?

Reply:
I've implemented this in the latest commit. There is no Bitmap::iter_from anymore, and the iterators do not use a wrapper in advance_to, and all uses binary search